pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/TransformOperandFactory.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.operator.operands;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.query.planner.logical.RexExpression;
+import org.apache.pinot.query.runtime.operator.utils.OperatorUtils;
+
+
+public class TransformOperandFactory {
+  private TransformOperandFactory() {
+  }
+
+  public static TransformOperand getTransformOperand(RexExpression rexExpression, DataSchema dataSchema) {
+    if (rexExpression instanceof RexExpression.FunctionCall) {
+      return getTransformOperand((RexExpression.FunctionCall) rexExpression, dataSchema);
+    } else if (rexExpression instanceof RexExpression.InputRef) {
+      return new ReferenceOperand(((RexExpression.InputRef) rexExpression).getIndex(), dataSchema);
+    } else if (rexExpression instanceof RexExpression.Literal) {
+      return new LiteralOperand((RexExpression.Literal) rexExpression);
+    } else {
+      throw new UnsupportedOperationException("Unsupported RexExpression: " + rexExpression);
+    }
+  }
+
+  private static TransformOperand getTransformOperand(RexExpression.FunctionCall functionCall, DataSchema dataSchema) {
+    List<RexExpression> operands = functionCall.getFunctionOperands();
+    int numOperands = operands.size();
+    String canonicalName = OperatorUtils.canonicalizeFunctionName(functionCall.getFunctionName());
+    switch (canonicalName) {
```

Comment:
Did we missing handle `IN` and `NOT IN`  or it's not required?

Reply:
Probably not required. They are either pushed down, or planned as semi join